.github/workflows/java-microservice-ci-default.yml

```diff
@@ -0,0 +1,679 @@
+name: CI for maven microservice
+
+on:
+  workflow_call:
+    inputs:
+      java_distribution:
+        description: "The java distribution (default to temurin)"
+        type: string
+        required: false
+        default: "temurin"
+      java_version:
+        description: "The java version (default to 17)"
+        type: string
+        required: false
+        default: "17"
+      helm_version:
+        description: "The helm version (default to 3.12.0)"
+        type: string
+        required: false
+        default: "3.12.0"
+      slack_channel_id_ci:
+        description: "The CI Slack channel's id"
+        type: string
+        required: true
+      slack_channel_id_ci_release:
+        description: "The CI release Slack channel's id"
+        type: string
+        required: true
+      sonar_organization:
+        description: "The sonar organization"
+        type: string
+        required: true
+      sonar_host_url:
+        description: "The sonar host url"
+        type: string
+        required: true
+      sonar_project_key:
+        description: "The sonar project key"
+        type: string
+        required: true
+      project_name:
+        description: "The project name"
+        type: string
+        required: true
+      aws_region:
+        description: "The AWS region"
+        type: string
+        required: false
+        default: "eu-west-1"
+      ecr_url:
+        description: "The ECR url"
+        type: string
+        required: true
+      chart_museum_url:
+        description: "The ChartMuseum URL"
+        type: string
+        required: true
+      enable_trivy_exit:
+        description: "Enable Trivy failure when CVE are found, you can deactivate this behavior"
+        type: boolean
+        required: false
+        default: true
+      enable_database_docker:
+        description: "Enable the database docker build"
+        type: boolean
+        required: false
+        default: true
+    secrets:
+      SLACK_BOT_TOKEN:
+        description: "The slack bot token"
+        required: true
+      SONAR_TOKEN:
+        description: "The sonar token"
+        required: true
+      AWS_ACCESS_KEY_ID:
+        description: "The AWS access key"
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        description: "The AWS secret access key"
+        required: true
+
+jobs:
```

Comment:
Maybe we can try to split in some more jobs.

Take a look at https://github.com/liquibase/liquibase/blob/master/.github/workflows/build.yml

I know artifacts sharing is not possible with docker images for us, but maybe we can find interesting things in this build.